symFileManager.py

```diff
@@ -48,38 +50,46 @@ def GetLibSymbolMap(self, libName, breakpadId):
     if libName == "":
       return None
 
+    # Guess the name of the .sym file on disk
+    if libName[-4:] == ".pdb":
+      symFileName = re.sub(r"\.[^\.]+$", ".sym", libName)
+    else:
+      symFileName = libName + ".sym"
+
+    pathSuffix = os.path.join(libName, breakpadId, symFileName)
+    urlSuffix = "/".join([libName, breakpadId, symFileName])
+
     # Check cache first
     libSymbolMap = None
     if libName in self.sCache and breakpadId in self.sCache[libName]:
       libSymbolMap = self.sCache[libName][breakpadId]
       self.UpdateMruList(libName, breakpadId)
```

Comment:
Not sure i understand that argument, let's say the limit for in-memory cache is 3 and the limit for on-disk cache is 7.

So the LRU might look like this [a, b, c, d, e, f, g]. The elements in memory are lru[0:3] and the elements on disk are lru[:7]

If "g" get used, then the LRU looks like [g, a, b, c, d, e, f]. The memory cache is lru[:3] = [g, a, b]. 
That means "c" needs to be evicted from the memory-cache and "g" needs to be added in. The disk-cache is unchanged. It's the same behaviour as now.

if "h" gets added, then LRU looks like [h, g, a, b, c, d, e] which means "b" gets evicted from in-memory-cache, "f" gets evicted from disk cache, and "h" gets added to both the memory-cache and the disk-cache.

I don't think it's necessary to worry too much about data structure access/iteration efficiency during eviction. The LRU is going to have fewer than 1000 elements in the memory cache (representing 1000 files cached  in memory) and you won't need to iterate beyond that.